db-scheduler/src/main/java/com/github/kagkarlsson/scheduler/ExecutePicked.java

```diff
@@ -78,22 +77,20 @@ private void executePickedExecution(Execution execution) {
         }
 
         Instant executionStarted = clock.now();
-        try {
-            LOG.debug("Executing " + execution);
-            CompletionHandler completion = task.get().execute(execution.taskInstance, new ExecutionContext(schedulerState, execution, schedulerClient));
+        CompletableFuture<CompletionHandler> completableFuture = task.get().execute(execution.taskInstance, new ExecutionContext(schedulerState, execution, schedulerClient));
+        completableFuture.whenComplete((completion, ex) -> {
+            LOG.debug("Completing " + execution + " on " + Thread.currentThread().getName());
+            if (ex != null) {
```

Comment:
Lets keep The exceptionhandling code (added vs removed) is  equivalent ?
ie, keep it same, catch runtimeexception and throwables.